### PR TITLE
Added typings for xml-formatter

### PR DESCRIPTION
**Pull request by holwech**
Please fill in this template.

- [x ] Use a meaningful title for the pull request. Include the name of the package modified.
- [x ] Test the change in your own code. (Compile and run.)
- [x ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/xml-formatter/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for xml-formatter 1.1
+// Project: https://github.com/chrisbottin/xml-formatter/ (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (amcasey) [inline types/xml-formatter/index.d.ts]: You can remove the parenthesized hint text.

File: types/xml-formatter/xml-formatter-tests.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import format = require("xml-formatter");
+
+const testXml = '<text><test><test><text>';
```
Comment (amcasey) [inline types/xml-formatter/xml-formatter-tests.ts]: It looks like some of these might have been intended to be close tags.

File: types/xml-formatter/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/*~ If you want to expose types from your module as well, you can
+ *~ place them in this block. Often you will want to describe the
+ *~ shape of the return type of the function; that type should
+ *~ be declared in here, as this example shows.
+ */
+declare namespace Format {
+    interface Options {
+        debug?: boolean;
+        indetation?: string;
```
Comment (amcasey) [inline types/xml-formatter/index.d.ts]: "indentation"?

File: types/xml-formatter/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+// Type definitions for xml-formatter 1.1
+// Project: https://github.com/chrisbottin/xml-formatter/ (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Joachim Holwech <https://github.com/holwech>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/*~ Note that ES6 modules cannot directly export callable functions.
```
Comment (amcasey) [inline types/xml-formatter/index.d.ts]: These comments look like they might be from a template.  Consider updating them with content specific to xml-formatter.

**Review by amcasey — CHANGES_REQUESTED** [review]

**Comment by holwech** [comment]
Thanks, fixed now

File: types/xml-formatter/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+// Type definitions for xml-formatter 1.1
+// Project: https://github.com/chrisbottin/xml-formatter/
+// Definitions by: Joachim Holwech <https://github.com/holwech>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace xmlFormatter;
```
Comment (amcasey) [inline types/xml-formatter/index.d.ts]: Sorry for the delay - I lost track of this one.  Is this namespace export necessary?  It's usually used to expose something in the global namespace (i.e. for webpages) and the underlying package doesn't appear to do that.

Comment (holwech) [reply types/xml-formatter/index.d.ts]: Yeah, you are probably right. I'm not an expert on this, I can remove it if you think it shouldn't be there.

Comment (amcasey) [reply types/xml-formatter/index.d.ts]: I'd leave it out for now.  It's easy to extend typings later. :smile:

**Review by amcasey — APPROVED** [review]

**Pull request merged**